test/integration/expression_test.rb

```diff
@@ -29,18 +29,24 @@ def test_float
   def test_range
     assert_equal(1..2, parse_and_eval("(1..2)"))
     assert_equal(3..4, parse_and_eval(" ( 3 .. 4 ) "))
+    assert_equal(0..0, parse_and_eval("('a'..'b')"))
+
+    with_error_mode(:strict) do
+      e = assert_raises(NoMethodError) { parse_and_eval("(1..(1..5))") }
+      assert_match(/undefined method `to_i' for 1..5:Range/, e.message)
+    end
   end
 
   private
 
   def parse_and_eval(markup, **assigns)
     if Liquid::Template.error_mode == :strict
       p = Liquid::Parser.new(markup)
-      markup = p.expression
-      p.consume(:end_of_string)
+      p.expression
+    else
+      expression = Liquid::Expression.parse(markup)
+      context = Liquid::Context.new(assigns)
+      context.evaluate(expression)
     end
```

Comment:
This is parsing without evaluating in strict parse mode.

```suggestion
      expression = p.expression
    else
      expression = Liquid::Expression.parse(markup)
    end
    context = Liquid::Context.new(assigns)
    context.evaluate(expression)
```